src/files/entities/extractions/docling-extraction.entity.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Embeddable, Property } from '@mikro-orm/core';
+
+import { Extraction } from './extraction.entity';
+
+import { ExtractionBackend } from '@/files/extraction/constants';
+
+@Embeddable({ discriminatorValue: ExtractionBackend.DOCLING })
+export class DoclingExtraction extends Extraction {
+  backend = ExtractionBackend.DOCLING;
+
+  @Property()
+  storageId?: string;
```

Comment:
Note that you're free to adjust the structure as you see fit. E.g. there can be 3 files, one with serialized docling document, one with markdown and one with chunks.